executor/cluster_reader.go

```diff
@@ -36,7 +36,7 @@ import (
 )
 
 type clusterRetriever interface {
-	retrieve(ctx sessionctx.Context) ([][]types.Datum, error)
+	retrieve(sctx sessionctx.Context, ctx context.Context) ([][]types.Datum, error)
```

Comment:
Please place the `context.Context` before `sessionctx.Context`.

Reply:
great, done.